src/Lykke.Service.ClientSearch/Startup.cs

```diff
@@ -0,0 +1,171 @@
+                .AddEnvironmentVariables();
+            Configuration = builder.Build();
+
+            Environment = env;
+        }
+
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc(x =>
+            {
+                if (!Environment.IsDevelopment())
+                {
+                    var authenticatedUserPolicy = new AuthorizationPolicyBuilder()
+                        .RequireAuthenticatedUser()
+                        .Build();
+
+                    x.Filters.Add(new AuthorizeFilter(authenticatedUserPolicy));
+                }
+            })
+                .AddJsonOptions(options =>
+                 {
+                     options.SerializerSettings.ContractResolver = new Newtonsoft.Json.Serialization.DefaultContractResolver();
+                 });
+
+            services.AddSwaggerGen(options =>
+            {
+                options.DefaultLykkeConfiguration("v1", "ClientSearch API");
+            });
+
+            var builder = new ContainerBuilder();
+
+            var appSettings = Configuration.LoadSettings<AppSettings>();
+
+            var log = CreateLogWithSlack(services, appSettings);
+
+            builder.RegisterModule(new ServiceModule(log));
+
+            builder.RegisterInstance<IPersonalDataService>(new PersonalDataService(appSettings.CurrentValue.PersonalDataServiceClient, log));
+
+            builder.AddTriggers(pool =>
+            {
+                pool.AddDefaultConnection(appSettings.CurrentValue.ClientSearchService.ClientPersonalInfoConnString);
+            });
+
+            builder.Populate(services);
+            ApplicationContainer = builder.Build();
+
+
+            return new AutofacServiceProvider(ApplicationContainer);
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplicationLifetime appLifetime, IPersonalDataService personalDataService, ILog log)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.Use((context, next) =>
+            {
+                if (!PersonalDataLoader.indexCreated)
+                {
+                    using (var writer = new StreamWriter(context.Response.Body))
+                    {
+                        context.Response.StatusCode = 503;
+                        return context.Response.WriteAsync("Search index is not ready yet");
+                    }
+                }
+
+                return next();
+            });
+
+
+            app.UseLykkeMiddleware("ClientSearch", ex => new { Message = "Technical problem" });
+
+            app.UseMvc();
+            app.UseSwagger();
+            app.UseSwaggerUi();
+
+
+            appLifetime.ApplicationStopped.Register(() =>
+            {
+                ApplicationContainer.Dispose();
+            });
+
+            Task task = Task.Factory.StartNew(() => {

```

Comment:
Do not execute here any initialization. Use a Statup manager as in the standard template https://github.com/LykkeCity/lykke.dotnettemplates/blob/0b506d6d6326a558c7107c40c4f78f7ae1f5bac0/Lykke.Service.LykkeService/src/Lykke.Service.LykkeService/Startup.cs#L109

Reply:
Initialization code was moved to StartupManager.